api/v1alpha1/nodepool_types.go

```diff
@@ -150,10 +154,10 @@ type NodePoolManagement struct {
 type NodePoolAutoScaling struct {
 	// +optional
```

Comment:
Is this still +optional without the pointer?

Reply:
this should actually be required when `NodePoolAutoScaling` is set, I'll update.